### PR TITLE
add unit tests for AddressService

### PR DESCRIPTION
**Pull request by joajabn**

### CODE REVIEW

**Comment by joajabn** [comment]
Questions which came across my mind during coding:
Should I also add a test shouldReturnEmptyList? (I'm not sure it's needed)
Should I test a method returnAddressIfExists from AddressService? (I believe no but I prefer to ask :))

File: src/test/java/pl/jablonskanycz/bakery/database/services/AddressServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    @Autowired
+    private AddressService addressService;
+
+    @MockBean
+    private AddressRepository addressRepository;
+
+    @Test
+    public void shouldReturnAllAddresses() {
+        //given
+        AddressEntity address1 = AddressEntity.builder().addressId(1L).latitude(51.20).longitude(15.80).build();
```
Comment (PaniOdProgramowania) [inline src/test/java/pl/jablonskanycz/bakery/database/services/AddressServiceTest.java]: do we have any lines that repeat within this file that we could extract to a method:)?

File: src/test/java/pl/jablonskanycz/bakery/database/services/AddressServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        //given
+        AddressEntity address = AddressEntity.builder().latitude(51.20).longitude(15.80).build();
+        when(addressRepository.save(any(AddressEntity.class))).thenReturn(address);
+
+        //when
+        addressService.addAddress(51.20, 15.80);
+
+        //then
+        verify(addressRepository).save(any(AddressEntity.class));
+    }
```
Comment (PaniOdProgramowania) [inline src/test/java/pl/jablonskanycz/bakery/database/services/AddressServiceTest.java:92]: in such a test it a good choice to test if a specific method was called. In more sophisticated tests we would probably ant to rather test the _outcome_ of the operation, not what was called underneath (not to test the inner implementation but to test the outcome).

File: src/test/java/pl/jablonskanycz/bakery/database/services/AddressServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@ActiveProfiles("test")
+class AddressServiceTest {
```
Comment (PaniOdProgramowania) [inline src/test/java/pl/jablonskanycz/bakery/database/services/AddressServiceTest.java:25]: when you run the tests, do they pass :)?

**Review by PaniOdProgramowania — COMMENTED** [review]
Good job! We will use it on our tomorrow's class :)

**Review by PaniOdProgramowania — APPROVED** [review]
🎉

**Pull request merged**